### ISSUE TITLE
improve add(Re)Issuance API

### ISSUE THREAD

**Issue by tiero** [vulpemventures/liquidjs-lib#88]
At the moment we force uset to specify one address/amount for each asset issued and token. We should allow to issue and reissue asset to multiple outputs in the same tx.

the API could either change:

1) accepting an array of recipients (address/script & amount in `addInIssuance` & `addInReissuance`
2) having the user manually add the asset & token outputs, the `addInIssuance` & `addInReissuance` only care about adding entropy & assetAssetBlinder?

Toughts @altafan @louisinger

**Comment by tiero** [comment]
I would go for the second approach: is a bit counter-intuitive the fact the addIn(Re)Issuance methods also add outputs: better to let the user add the asset & token output, eventually if user forgots, we should throw an instructive errore message

**Comment by altafan** [comment]
The API is designed like this to prevent the user to make mistakes when adding outputs. This stands ONLY for issuances:
to generate the correct asset hash of the reissuance token, the user would have to use the correct bool value that expresses whether the issuance is blinded or not.

I personally prefer the first approach, because I consider adding an issuance/resissuance to a tx not just adding stuff to inputs, but also adding some new output.

Now that we are discussing this, I think the correct name for these APIs are `addIssuance` and `addReissuance` instead of `addInIssuance` and `addInReissuance`, without any specific reference to inputs. We are still in time to make this change if we go for 1)

**Comment by tiero** [comment]
> prevent the user to make mistakes when adding outputs

What if we give some convenience method out of the Pset class that introspect the input and generates the right asset?


some pseudocode?

```js 
const updater = new Updater(ptx)
updater.addInput({ reissuanceOpts: { tokenBlinder, entropy, intialBlindedIssuance }})
updater.addOutput({ asset: ptx.issuance.asset })
updater.addOutput({ asset: ptx.issuance.token })

```

**Comment by altafan** [comment]
Don't really like this solution mostly because there's no asset filed in the input issuance.

Another option could be to have convenience methods AddIssuance/AddReissuance (those already existing), and AddInIssuance/AddInReissuance to add only the input part of the issuance, leaving the output one to the user, like experienced users, what do you think?

**Comment by tiero** [comment]
Let's go for `AddIssuance/AddReissuance` as a method of the updater, where you can pass an array of assetOutputs & outputs? We basically phase out assetAddress & amountAddress + tokenAddress & tokenAmount  

```js
updater.addReissuance({
  assetOutputs,
  tokenOutputs
})
```

**Comment by altafan** [comment]
Agreed.

**Comment by altafan** [comment]
cc @louisinger

TLDR: let's rename `addInIssuance | addInReissuance` to `addIssuance | addReissuance` and expect an array of asset/token outputs defined as `[{address, amount}]` instead of `assetAddress | amountAddress` and `tokenAddress | tokenAmount` in the respective args.
